File: ruby/bob/bob.rb
class Bob
  def self.hey string
    testString = string.strip
    return 'Fine. Be that way!' if testString.empty?
    return 'Whoa, chill out!' if testString.upcase! == nil && testString.match(/[a-zA-Z]/)
    return 'Sure.' if testString.end_with?('?')
    'Whatever.'
  end
end

module BookKeeping
  VERSION = 1
end

File: python/bob/bob.py
from re import match

def hey(string):
    if string.isupper():
        return 'Whoa, chill out!'
    elif string.strip() == '':
        return 'Fine. Be that way!'
    elif string.strip()[-1] == '?':
        return 'Sure.'
    else:
        return 'Whatever.'